### PR TITLE
solana/gateway_v2: Add utils for getting fee and working out fee

### PR DESCRIPTION
**Pull request by JulianSpring**
Added util functions for fees.

### CODE REVIEW

File: solana/program_v2/programs/gateway_v2/src/util.rs [hunk 1]
```diff
@@ -23,3 +26,101 @@ pub trait OnChainSizeWithArg<Arg> {
...
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+/// get_fees
+/// Returns two fees in the correct unit
+/// First result returns the fee for the guardian
+/// Second result returns the gatekeeper fee
+pub fn work_out_fees(fee: u64, split: u16) -> (u64, u64) {
+    let percentage = (split as f64).div(100_f64);
+    let guardian_fee = (fee as f64).mul(percentage);
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/util.rs:44]: Let's call this `network_fee` ? the idea of `gaurdian` doesn't really exist outside of tests/implementations... 

I'd think of `gaurdian fee` as a collection of `network fees` (as it's 1:n from gaurdian to network)

Comment (JulianSpring) [reply solana/program_v2/programs/gateway_v2/src/util.rs:44]: That makes sense, will update now.

File: solana/program_v2/programs/gateway_v2/src/util.rs [hunk 2]
```diff
@@ -23,3 +26,101 @@ pub trait OnChainSizeWithArg<Arg> {
...
+
+pub fn get_gatekeeper_fees(fees: &[GatekeeperFees], mint: Pubkey) -> &GatekeeperFees {
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+pub fn get_network_fees(fees: &[NetworkFees], mint: Pubkey) -> &NetworkFees {
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+/// get_fees
```
Comment (JulianSpring) [inline solana/program_v2/programs/gateway_v2/src/util.rs:38]: Also need to remove this 😓

File: solana/program_v2/programs/gateway_v2/src/util.rs (hunk 1 above)
Comment (JulianSpring) [reply solana/program_v2/programs/gateway_v2/src/util.rs]: Done!

File: solana/program_v2/programs/gateway_v2/src/util.rs [hunk 3]
```diff
@@ -23,3 +26,101 @@ pub trait OnChainSizeWithArg<Arg> {
...
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+pub fn get_network_fees(fees: &[NetworkFees], mint: Pubkey) -> &NetworkFees {
+    fees.iter().find(|&&x| x.token == mint).unwrap()
+}
+
+/// calculate_network_and_gatekeeper_fee
+/// Returns two fees in the correct unit
+/// First result returns the fee for the guardian
```
Comment (william-brooks) [inline solana/program_v2/programs/gateway_v2/src/util.rs:40]: Just guardian in the comment :)

**Review by william-brooks — APPROVED** [review]

**Pull request merged**